src/relayclient/RelayClient.ts

```diff
@@ -440,6 +430,66 @@ export class RelayClient {
       message: `Transaction was not audited. Failed audit calls: ${failedAuditorsCount}/${auditors.length}`
     }
   }
+
+  getUnderlyingProvider (): HttpProvider {
+    return this.rawConstructorInput.provider
+  }
+
+  async resolveConfiguration ({
+    provider,
+    partialConfig = {},
+    overrideDependencies = {}
+  }: GSNUnresolvedConstructorInput): Promise<void> {
+    const versionManager = new VersionsManager(gsnRuntimeVersion, partialConfig.requiredVersionRange)
+    const contractInteractor = overrideDependencies?.contractInteractor ??
+      await new ContractInteractor({
+        provider,
+        versionManager,
+        logger: this.logger,
+        deployment: { paymasterAddress: partialConfig?.paymasterAddress }
+      }).init()
+
+    const isMetamask: boolean = (provider as any).isMetaMask
+
+    // provide defaults valid for metamask (unless explicitly specified values)
+    const methodSuffix = partialConfig.methodSuffix ?? (isMetamask ? '_v4' : defaultGsnConfig.methodSuffix)
+    const jsonStringifyRequest = partialConfig.jsonStringifyRequest ?? (isMetamask ? true : defaultGsnConfig.jsonStringifyRequest)
+
+    const resolvedConfig: Partial<GSNConfig> = {
+      methodSuffix,
+      jsonStringifyRequest
+    }
+    this.config = {
+      ...defaultGsnConfig,
+      ...resolvedConfig,
+      ...partialConfig
+    }
+
+    const accountManager = overrideDependencies?.accountManager ?? new AccountManager(provider, contractInteractor.chainId, this.config)
```

Comment:
I think that even as internals, resolve config and resolve deps should be separate functions.

Reply:
Separated them, no problem.